provider/pkg/schema/resource_props.go

```diff
@@ -0,0 +1,31 @@
+package schema
+
+import (
+	"strings"
+
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/naming"
+)
+
+// Represents a property listed in `readOnlyProperties`, `writeOnlyProperties`,
+// or `createOnlyProperties` in the Cloud Control schema.
+type ResourceProperty string
+
+// ToSdkName converts the property name to an SDK-compatible name.
+// This will also handle converting all nested properties.
+//   - Nested object properties are delimited with `/`
+//   - Nested array properties are delimited with `/*/`
+//
+// e.g.
+//   - `Foo/Bar/Baz` => `foo/bar/baz`
+//   - `Foo/*/BarBaz` => `foo/*/barBaz`
+func (r *ResourceProperty) ToSdkName() string {
+	arrayProps := strings.Split(string(*r), "/*/")
```

Comment:
I think this would panic right now when called on an `r` that's nil.

Reply:
`contract.Assertf(r != nil, "ResourceProperty should not be nil")`

panics are good sometimes - this should not be nil I think.

Reply:
I'd argue that the provider shouldn't crash on a panic and rather exit with a proper error message.
If this shouldn't be nil, then we shouldn't accept a nil as the input imo

Reply:
That's a better idea. I checked around and this seems fine:

```func (r ResourceProperty) ToSdkName() string```

Reply:
If you want to be more pedantic, which is not always bad, then this:

```
n := ResourceProperty(strings.TrimPrefix(v.(string), "/properties/"))
```

Should instead be:

```
parseResourceProperty(string) (ResourceProperty, error)
```

Reply:
Switched it to not be a pointer receiver.